src/fhirtypes/ElementDefinition.ts

```diff
@@ -602,6 +623,301 @@ export class ElementDefinition {
+    ) {
+      this.fixedInstant = value;
+    } else if (
+      type === 'date' &&
+      /^([0-9]([0-9]([0-9][1-9]|[1-9]0)|[1-9]00)|[1-9]000)(-(0[1-9]|1[0-2])(-(0[1-9]|[1-2][0-9]|3[0-1]))?)?$/.test(
+        value
+      ) &&
+      this.checkIfFixable(value, this.fixedDate, type)
+    ) {
+      this.fixedDate = value;
+    } else if (
+      type === 'dateTime' &&
+      /^([0-9]([0-9]([0-9][1-9]|[1-9]0)|[1-9]00)|[1-9]000)(-(0[1-9]|1[0-2])(-(0[1-9]|[1-2][0-9]|3[0-1])(T([01][0-9]|2[0-3]):[0-5][0-9]:([0-5][0-9]|60)(\.[0-9]+)?(Z|(\+|-)((0[0-9]|1[0-3]):[0-5][0-9]|14:00)))?)?)?$/.test(
+        value
+      ) &&
+      this.checkIfFixable(value, this.fixedDateTime, type)
+    ) {
+      this.fixedDateTime = value;
+    } else if (
+      type === 'time' &&
+      /^([01][0-9]|2[0-3]):[0-5][0-9]:([0-5][0-9]|60)(\.[0-9]+)?$/.test(value) &&
+      this.checkIfFixable(value, this.fixedTime, type)
+    ) {
+      this.fixedTime = value;
+    } else if (
+      type === 'oid' &&
+      /^urn:oid:[0-2](\.(0|[1-9][0-9]*))+$/.test(value) &&
+      this.checkIfFixable(value, this.fixedOid, type)
+    ) {
+      this.fixedOid = value;
+    } else if (
+      type === 'id' &&
+      /^[A-Za-z0-9\-\.]{1,64}$/.test(value) &&
+      this.checkIfFixable(value, this.fixedId, type)
+    ) {
+      this.fixedId = value;
+    } else if (
+      type === 'markdown' &&
+      /^\s*(\S|\s)*$/.test(value) &&
+      this.checkIfFixable(value, this.fixedMarkdown, type)
+    ) {
+      this.fixedMarkdown = value;
+    } else if (type === 'uuid' && this.checkIfFixable(value, this.fixedUuid, type)) {
+      this.fixedUuid = value;
+    } else {
+      throw new MismatchedTypeError('string', value, type);
+    }
+  }
+
+  /**
+   * Fixes a Quantity to this element.
+   * @see {@link fixValue}
+   * @param {FshQuantity} value - the Quantity value to fix
+   * @throws {NoSingleTypeError} when the ElementDefinition does not have a single type
+   * @throws {ValueAlreadyFixedError} when the value is already fixed to a different value
+   * @throws {TypeNotFoundError} when the value does not match the type of the ElementDefinition
+   */
+  fixFshQuantity(value: FshQuantity): void {
+    if (!this.hasSingleType()) {
+      throw new NoSingleTypeError('Quantity');
+    }
+    const type = this.type[0].code;
+    if (type === 'Quantity') {
+      if (this.patternQuantity) {
+        const found = this.patternQuantity;
+        const foundFshQuantity = new FshQuantity(
+          found.value,
+          new FshCode(found.code, found.system)
+        );
+        // Check if the new quantity matches the current
+        if (!value.equals(foundFshQuantity)) {
+          throw new ValueAlreadyFixedError(
+            value.toString(),
+            'Quantity',
+            foundFshQuantity.toString()
+          );
+        }
+        // if they do match, there is nothing to do, so return
+        return;
+      }
+      if (value.unit) {
+        this.fixFshCode(value.unit);
```

Comment:
Oh nice.  I forgot that I implemented `fixedCode` for `Quantity`.  That said, for a more casual reader of the code, this line might be confusing.  I'd consider adding a comment indicating that the `fixFshCode` has special support for fixing `Quantity` units as codes.